src/include/duckdb/execution/operator/persistent/csv_scanner/csv_buffer_manager.hpp

```diff
@@ -0,0 +1,105 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/execution/operator/persistent/csv_scanner/csv_buffer_manager.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/main/client_context.hpp"
+#include "duckdb/execution/operator/persistent/csv_scanner/csv_file_handle.hpp"
+#include "duckdb/execution/operator/persistent/csv_scanner/csv_reader_options.hpp"
+
+namespace duckdb {
+class CSVBuffer;
+class CSVStateMachine;
+//! This class is used to manage the buffers
+//! Buffers are cached when used for auto detection
+//! Otherwise they are not cached and just returned
```

Comment:
What are these buffers, and what's in them? Parts of a CSV file? Let's extend the class description a bit. :)